front/app/components/admin/PostManager/index.tsx

```diff
@@ -122,44 +124,48 @@ interface Props extends InputProps, DataProps {}
 export type TFilterMenu = 'topics' | 'phases' | 'projects' | 'statuses';
 export type PreviewMode = 'view' | 'edit';
 
-interface State {
-  /** A set of ids of ideas/initiatives that are currently selected */
-  selection: Set<string>;
-  activeFilterMenu: TFilterMenu;
-  searchTerm: string | undefined;
-  previewPostId: string | null;
-  previewMode: PreviewMode;
-}
-
-export class PostManager extends React.PureComponent<Props, State> {
-  constructor(props: Props) {
-    super(props);
-    this.state = {
-      selection: new Set(),
-      activeFilterMenu: props.defaultFilterMenu,
-      searchTerm: undefined,
-      previewPostId: null,
-      previewMode: 'view',
-    };
-  }
-
-  componentDidUpdate(prevProps: Props) {
-    const { visibleFilterMenus } = this.props;
-
-    if (
-      prevProps.visibleFilterMenus !== visibleFilterMenus &&
-      !visibleFilterMenus.find(
-        (item) => item === this.state.activeFilterMenu
-      ) &&
-      visibleFilterMenus[0]
-    ) {
-      this.setState({ activeFilterMenu: visibleFilterMenus[0] });
+const PostManager = ({
```

Comment:
Finally converted the PostManager to a functional component - couldn't resist anymore 😂 There's still some work to be done in removing all the hocs but we can do this down the line.